code/datums/components/holderloving.dm

```diff
@@ -0,0 +1,48 @@
+/// Prevents items from leaving their storage or whoever is holding their storage (does not need to have a storage component)
+/datum/component/holderloving
+	can_transfer = TRUE
+	/// Item that is kept track of
+	var/obj/item/holder
+	/// If we delete our parent when the holder gets deleted
+	var/del_parent_with_holder = FALSE
+
+/datum/component/holderloving/Initialize(_holder, _del_parent_with_holder)
+	if(!isitem(parent) || !_holder)
+		return COMPONENT_INCOMPATIBLE
+	holder = _holder
+	del_parent_with_holder = (_del_parent_with_holder || FALSE)
+
+/datum/component/holderloving/RegisterWithParent()
+	RegisterSignal(holder, COMSIG_MOVABLE_MOVED, .proc/check_my_loc)
+	RegisterSignal(holder, COMSIG_PARENT_QDELETING, .proc/holder_deleting)
+	RegisterSignal(parent, list(	COMSIG_ITEM_DROPPED, COMSIG_ITEM_EQUIPPED, \
+									COMSIG_STORAGE_ENTERED, COMSIG_STORAGE_EXITED ), .proc/check_my_loc)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#tabs-not-spaces

Only one indentation level, and remove the space at the end.